### PR TITLE
Fix error in combineWidgets when no widgets given

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue #3618

## Relevant technical choices

- See comments below

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/googlesitekit/widgets/util/combine-widgets.test.js [hunk 1]
```diff
@@ -232,4 +232,13 @@ describe( 'combineWidgets', () => {
...
 		);
 	} );
+
+	it( 'works with no widgets', () => {
+		const widgets = [];
+		const widgetStates = [];
+		const layout = getWidgetLayout( widgets, widgetStates );
+		expect( () =>
+			combineWidgets( widgets, widgetStates, layout )
+		).not.toThrow();
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/widgets/util/combine-widgets.test.js:242]: You can see this test fail with the error in the issue before the early returns were added here https://github.com/google/site-kit-wp/issues/3618#issuecomment-892488288

File: assets/js/googlesitekit/widgets/util/get-widget-layout.js [hunk 2]
```diff
@@ -119,12 +119,15 @@ function getNextActiveWidget( offset, widgets, widgetStates ) {
  *                  each active widget.
  */
 export function getWidgetLayout( widgets, widgetStates ) {
-	let counter = 0;
-	let rowIndex = 0;
-
 	let columnWidths = [];
 	const rowIndexes = [];
 
+	if ( ! widgets?.length ) {
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/widgets/util/get-widget-layout.js:125]: This return wasn't necessary to fix this but seemed like a good idea to add.

**Review by aaemnnosttv — COMMENTED** [review]
The problem was due to `shouldCombineAllWidgets` defaulting to `true` and assuming there are always 1 or more widgets given. Neither return value really makes sense when there are no widgets though because there is nothing to combine, so the condition was added to return early before it got to that point.

**Review by tofumatt — APPROVED** [review]

**Comment by tofumatt** [comment]
Not sure why the tests aren't running here, but running them locally seemed good 👍🏻

**Pull request merged**